src/core/state.py

```diff
@@ -107,13 +109,9 @@ class StateTime:
     This class associates the state with the time.
```

Comment:
Might want to add a `__str__` method to better format the printing of the `StateTime` object. Currently, its hard to discern the CLI logging information in real time

Reply:
All `dataclass`es already come with a pre-built `__str__`. It is intentionally verbose so that we can debug later - no need to be able to read it in real-time

Reply:
True, we wouldn't be reading from the CLI, we'd be reading from the CSVs